_posts/2019-11-06-machine-readable-output.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Machine-readable Output Formats
+author: José Iván López González
+layout: post
+piwik: true
+---
+
+Third party programs and scripts (e.g., YaST) parse the output of the *snapper* cli commands to get some information. For example, to find out the number of all pre and post snapshots, scripts can run the following:
+
+~~~
+snapper list | awk '/pre/||/post/{print $3}'
+~~~
+
+Output from *snapper list* command is a table intended to be read by humans but not scripts. This could make difficult to parse snapper outputs. But even worse, every change in this table output (e.g., by adding a new column or changing their values format) could ruin scripts that relied on the previous table version.
+
+~~~
+snapper --iso list --disable-used-space
+ # | Type   | Pre # | Date                | User | Cleanup | Description           | Userdata
+---+--------+-------+---------------------+------+---------+-----------------------+--------------
+0  | single |       |                     | root |         | current               |
+1* | single |       | 2019-10-15 12:45:25 | root |         | first root filesystem |
+2  | single |       | 2019-10-15 12:51:06 | root | number  | after installation    | important=yes
+3  | pre    |       | 2019-10-15 13:06:08 | root | number  | zypp(zypper)          | important=no
+4  | post   |     3 | 2019-10-15 13:07:41 | root | number  |                       | important=no
+~~~
+
+All that could make quite error prone to work with *snapper* cli output, and for that reason, *snapper* now offers some new options to generate listing outputs with CSV or JSON format.
```

Comment:
* cli -> CLI
* "and, for that reason," (if you remove the part within commas -for that reason- the sentence still makes sense).
* I would simplify the second part: "to use CSV or JSON for lists".